core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/Changeset.java

```diff
@@ -348,7 +377,28 @@ protected void setChangeset(Changeset from) {
 		this.statementCleared = from.statementCleared;
 	}
 
+	public static Changeset simpleClone(Changeset from) {
+		assert !from.closed;
+
+		Changeset changeset = new Changeset() {
+			@Override
+			public void flush() throws SailException {
+
+			}
+
+			@Override
+			public Model createEmptyModel() {
+				return from.createEmptyModel();
+			}
+		};
+
+		changeset.setChangeset(from);
+
+		return changeset;
+	}
```

Comment:
Since closing a changeset didn't previously do much, I found a few bugs where a changeset should be closed but we also needed a copy to keep working on.